app/javascript/availability/components/location_info.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import availability from '../index.js';
+
+const Spinner = () => (
+    <span className="spinner-border spinner-border-sm" 
+          role="status" aria-hidden="true"></span>
+);
+
+const LocationInfo = (props) => {
+    const holding = props.holding;
+    
+    // Location information presented to the user is different based on a few scenarios
+    // First, if it's related to ILL
+    if (availability.isIllLink(holding)) {
+        return (
+            <a 
+                data-type="ill-link" 
+                data-catkey={holding.catkey}
+                data-call-number={holding.callNumber} 
+                data-link-type={availability.illLinkType(holding)}
+                data-item-location={holding.locationID}
+                href="#"
+            >
+                <Spinner />
+                {availability.illLinkText(holding)}
+            </a>
+        )
+    }
+
+    // AEON
+    if (availability.isArchivalThesis(holding)) {
+        const illiadURL = "https://psu.illiad.oclc.org/illiad/upm/lending/lendinglogon.html";
+        const aeonLocationText = availability.mapLocation(availability.allLocations, holding.locationID);
+
+        return (<>
+            <a 
+                data-type="ill-link" 
+                data-catkey={holding.catkey}
+                data-call-number={holding.callNumber}
+                data-link-type="archival-thesis" 
+                data-item-type={holding.itemTypeID}
+                href="#">
+                    <Spinner />
+                    Request Scan - Penn State Users
+            </a><br />
+
+            <a href={illiadURL}>Request Scan - Guest</a><br />
+
+            <a 
+                data-type="aeon-link" 
+                data-catkey={holding.catkey}
+                data-call-number={holding.callNumber}
+                data-link-type="archival-thesis"
+                data-item-type={holding.itemTypeID}
+                data-item-id={holding.itemID}
+                data-item-location={aeonLocationText}
+                href="#">
+                    <Spinner />
+                    View in Special Collections
+            </a>
+        </>)
+    }
+
+    // Special Collections
+    if (availability.isArchivalMaterial(holding)) {
+        const specialCollectionsText = availability.mapLocation(availability.allLocations, holding.locationID);
+
+        return (<>
+            {specialCollectionsText}<br /> 
+
+            <a 
+                data-type="aeon-link" 
+                data-catkey={holding.catkey}
+                data-call-number={holding.callNumber}
+                data-link-type="archival-material"
+                data-item-type={holding.itemTypeID}
+                data-item-id={holding.itemID}
+                data-item-location={specialCollectionsText}
+                href="#">
+                    <Spinner />
+                    Request Material
+            </a>
+        </>);
+    }
+
```

Comment:
I wonder if you can take this one step further and so something like:

``` js
if (availability.isIllLink(holding)) {
  return (
    <IllLink holding={holding} />
  )
}

if (availability.isArchivalThesis(holding)) {
  return (
    <IllLink holding={holding} />
    <AeonLink holding={holding} />
  )
  
if (availability.isArchivalMaterial(holding)) {
return (
    <AeonLink holding={holding} />
  )
}
```

Reply:
@awead - good call. I was thinking of saving that for a second stage of refactor but maybe it makes sense to just do this now. There is some more ILL/AEON logic in `availability/index.js` that can then be moved into these components in a later stage of the refactor.

Reply:
Up to you. If you want to "tie it off" now and save it for later, that's ok by me.

Reply:
I was exactly thinking the same thing like it would be very nice to see `<IllLink holding={holding} />` and `<AeonLink holding={holding} />` components. I am totally fine for these to be the next phase of the refactor since as Nick mentioned this PR has already good amount of changes.